docs/graph-algorithms/prims-algorithm.mdx

```diff
@@ -0,0 +1,255 @@
+---
+title: Prim's Algorithm
+slug: /graph-algorithms/prims-algorithm
+---
+
+## Introduction
+
+The Prim's algorithm is used to find minimum spanning tree. This was originally discovered by the Czech mathematician Vojtěch Jarník in 1930. However this algorithm is mostly known as Prim's algorithm after the American mathematician Robert Clay Prim, who rediscovered and republished it in 1957.
+
+The Prim's algorithm is a greedy algorithm. This algorithm is used to find a minimum spanning tree for a weighted undirected graph. This means it finds a subset of the edges that forms a tree that includes every vertex, where the total weight of all the edges in the tree is minimized.
+
+## Explanation
+
+* From a given undirected graph find the path such that it covers all the edges and has minimum total weight.
+
+* Given an undirected and connected graph ```G=(V, E) where, V = vertices; E = edges```, a spanning tree of graph `G` is a tree that includes every vertex and edge of graph `G`.
+
+* A `weight` is associated with every edge and the sum of all the `weights` is known as `cost`. A **minimum spanning tree** is a tree having ```least cost```.
+
+* **Prim's Algorithm** is used to find **minimum spanning tree**. The algorithm traverses through every edge and picks up ```least weight``` between two vertices.
+
+* The algorithm picks up the `weights` in such a way that the ```weights are least``` and also the `edges` picked do not form a loop between any `vertices`.
+
+## Quick Example :
+
+* Consider a graph with ```6 vertices and 9 edges```. Now let's try to find out the minimum spanning tree using Prim's Algorithm.
+![graph](https://user-images.githubusercontent.com/60662775/114315588-3f1aa600-9b1d-11eb-9399-2bb42d280e6c.png)
+
+* Now traverse through each edge and pick up the ```least weights``` such that they do not form a `loop`.
+
+![step1](https://user-images.githubusercontent.com/60662775/114315644-75582580-9b1d-11eb-8c5f-a620d6ac2810.png)
+
+![step2](https://user-images.githubusercontent.com/60662775/114315649-79844300-9b1d-11eb-9e0a-f6570fdf72b2.png)
+
+![step3](https://user-images.githubusercontent.com/60662775/114315656-7db06080-9b1d-11eb-94b6-c1055498e383.png)
+
+![step4](https://user-images.githubusercontent.com/60662775/114315668-83a64180-9b1d-11eb-9d7b-5e86e1e1dcab.png)
+
+![step5](https://user-images.githubusercontent.com/60662775/114315674-8acd4f80-9b1d-11eb-9f9a-6717bf3ff1ed.png)
```

Comment:
The diagrams look like poor screenshots from some text book or a PDF. I'm sorry I cannot accept a PR with these diagrams. It would be best to remove them and create an issue for diagrams to be added to this.

Reply:
Then what type of diagrams can be added??